nextjs-app/components/pages/faq/ContentWithFilter.tsx

```diff
@@ -0,0 +1,28 @@
+"use client";
+
+import TagFilter from "@/components/common/TagFilter/TagFilter";
+import { FAQ_FILTER_OPTIONS, IFilterOption } from "@/constants/filter-options";
+import { useState } from "react";
+
+const ContentWithFilter = () => {
+  const [selectedFilter, setSelectedFilter] =
+    useState<IFilterOption["key"]>("all");
+
+  const handleSelect = (selectedOption: IFilterOption["key"]) => {
+    setSelectedFilter(selectedOption);
+    console.log("Selected: ", selectedOption);
+  };
```

Comment:
```suggestion
const [selectedFilter, setSelectedFilter] =
    useState("all");

  const handleSelect = (selectedOption: string) => {
    setSelectedFilter(selectedOption);
    console.log("Selected: ", selectedOption);
  };
```

The same, for our case, its type is just string